### PR TITLE
[No QA] Small refactors.

### PR DESCRIPTION
**Pull request by parasharrajat**
@marcaaron  Please review.
### Details

1. Just realized that we are updating the state unnecessary in the Tooltips & added on-demand rendering.
2. Marc's comment https://github.com/Expensify/Expensify.cash/pull/2553#discussion_r625494362

### Tested On

- [x] Web
- [ ] Mobile Web
- [x] Desktop
- [ ] iOS
- [ ] Android

### CODE REVIEW

File: src/pages/home/report/ReportActionsView.js [hunk 1]
```diff
@@ -155,7 +155,7 @@ class ReportActionsView extends React.Component {
         const previousLastSequenceNumber = lodashGet(lastItem(prevProps.reportActions), 'sequenceNumber');
         const currentLastSequenceNumber = lodashGet(lastItem(this.props.reportActions), 'sequenceNumber');
         const shouldRecordMaxAction = Visibility.isVisible()
-            && !(this.props.isDrawerOpen && this.props.isSmallScreenWidth);
+            && (!this.props.isSmallScreenWidth || !this.props.isDrawerOpen);
```
Comment (marcaaron) [inline src/pages/home/report/ReportActionsView.js:158]: Can we please add a comment here and list out every situations where we will record the max action?

File: src/components/Tooltip/index.js [hunk 2]
```diff
@@ -174,7 +160,7 @@ class Tooltip extends PureComponent {
             tooltipTextStyle,
             pointerWrapperStyle,
             pointerStyle,
-        } = getTooltipStyles(
+        } = this.state.isReady ? getTooltipStyles(
```
Comment (marcaaron) [inline src/components/Tooltip/index.js:163]: Is this necessary? We are not using any of these styles anyway unless `this.state.isReady === true`.

**Review by marcaaron — CHANGES_REQUESTED** [review]

File: src/components/Tooltip/index.js [hunk 3]
```diff
@@ -12,6 +12,9 @@ class Tooltip extends PureComponent {
         super(props);
 
         this.state = {
+            // Is tooltip rendered?
+            isReady: false,
```
Comment (cead22) [inline src/components/Tooltip/index.js:16]: NAB but seems like `isRendered` would be a better name

File: src/components/Tooltip/index.js (hunk 2 above)
Comment (parasharrajat) [reply src/components/Tooltip/index.js:163]: I think it's better to just move this style calculations to `TooltipRenderedOnPageBody` as they are only used there.

**Comment by parasharrajat** [comment]
Updated. Thanks

**Review by marcaaron — APPROVED** [review]

**Pull request merged**